### PR TITLE
Simplify Artery remote deployment and make inbound-lanes=4 default, #21422

### PR DESCRIPTION
**Pull request by patriknw**
* DaemonMsgCreate is not a system message. We send it over the control
  stream because remote deployment process depends on message ordering
  for DaemonMsgCreate and Watch messages. That is all good.
* We also send DaemonMsgCreate over the ordinary message stream (all
  outbound lanes) so that the first ordinary message that is sent to
  the ref does not arrive before the actor is created. This is not needed,
  since the retried resolve in the Decoder will take care of that anyway.
* Inbound lanes were not covered, but not needed.
* Then the deduplication of DaemonMsgCreate messages in  RemoteSystemDaemon
  is not needed.
* Added some more tests for these things.

### CODE REVIEW

**Comment by patriknw** [comment]
I will run this in a repeat job.

**Comment by patriknw** [comment]
Refs #21422

**Comment by patriknw** [comment]
I'll get back to this after vacation. There are a few more things I would like to do here before merging.

**Comment by ktoso** [comment]
Okey! Marked as stashed for now then

**Comment by patriknw** [comment]
I will rebase and fix the remaining when partition hub is merged.

**Comment by lightbend-cla-validator** [comment]
Hi @nvollmar,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by patriknw** [comment]
This is ready for final review. I'm running it in a repeat job now.

**Comment by patriknw** [comment]
Don't merge this before https://github.com/akka/akka/pull/23562
The change of default to inbound-lanes=4 revealed that problem.

**Review by johanandren — APPROVED** [review]
LGTM

File: akka-remote/src/main/resources/reference.conf
Comment (2m) [inline akka-remote/src/main/resources/reference.conf]: Should this say `with inbound-lanes=1`?

**Review by 2m — APPROVED** [review]
LGTM with a question.

File: akka-remote/src/main/resources/reference.conf
Comment (patriknw) [reply akka-remote/src/main/resources/reference.conf]: right, thx

**Comment by patriknw** [comment]
@2m @johanandren I added reference docs for the lanes, please take a look at baea69b and see if it makes sense.

File: akka-docs/src/main/paradox/scala/remoting-artery.md
Comment (johanandren) [inline akka-docs/src/main/paradox/scala/remoting-artery.md]: practice

Comment (johanandren) [inline akka-docs/src/main/paradox/scala/remoting-artery.md]: practice

**Review by johanandren — APPROVED** [review]
Docs looks good sans one spelling error

**Comment by patriknw** [comment]
javac NPE

**Comment by patriknw** [comment]
PLEASE BUILD

**Comment by patriknw** [comment]
PLS BUILD

**Comment by patriknw** [comment]
Refs #21422

**Pull request merged**
